File: sorting-algorithms/selection_sort.cpp
#include <iostream>
using namespace std;

void swap(int *x, int *y)
{
    int t = *x;
    *x = *y;
    *y = t;
}

void selection_sort(int arr[], int len)
{
    for (int i = 0; i < len - 1; i++)
    {
        int min_idx = i;
        for (int j = i + 1; j < len; j++)
        {
            if (arr[j] < arr[min_idx])
            {
                min_idx = j;
            }
        }
        swap(&arr[min_idx], &arr[i]);
    }
}

void printArray(int arr[], int size)
{
    for (int i = 0; i < size; i++)
        cout << arr[i] << " ";
}

int main()
{
    int arr[] = {64, 25, 12, 22, 11};
    int len = *(&arr + 1) - arr;
    int size = sizeof(arr) / sizeof(arr[0]);
    selection_sort(arr, len);
    printArray(arr, size);
    return 0;
}

File: sorting-algorithms/selection_sort.nim
proc selection_sort[T](arr: var openArray[T]) =
    let n = arr.len
    for i in 0..<n:
        var min_index = i
        for j in (i+1)..<n:
            if arr[j] < arr[min_index]:
                min_index = j
        swap(arr[min_index], arr[i])
    echo "sorted array: ", arr

var arr = @[64, 25, 12, 22, 11]
echo "array is:", arr
selection_sort(arr)
